modules/internal/ChapelLocale.chpl

```diff
@@ -304,12 +304,6 @@ module ChapelLocale {
       return this;
     }
 
-    // A useful default definition is provided (not pure virtual).
-    pragma "no doc"
-    override proc writeThis(f) throws {
-      f <~> name;
-    }
-
```

Comment:
Why are these moved to ChapelIO? It is OK for internal types but I'd just like to know user types / inheritance hierarchies can define these on their own.

Reply:
Having them declared here causes a circular dependency, since IO relies on locales.  Moving the writeThis function to ChapelIO allows us to wait to resolve those methods until the locales have been initialized (so it would be safe to initialize IO).